### PR TITLE
Enable prompts skip for ci

### PR DESCRIPTION
**Pull request by louiszawadzki**
### What and why?

Enable to skip prompts for CI purposes.
This will enable us to E2E test this wizard with some react native apps nightly.

### How?

Enable to pass prompted arguments through command arguments.

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit)

### CODE REVIEW

File: src/commands/setup/interface.ts [hunk 1]
```diff
@@ -1 +1,5 @@
-export type DatadogSite = "US" | "EU" | "US3" | "US5" | "GOV";
+export const datadogSites = ["US", "EU", "US3", "US5", "GOV"] as const;
```
Comment (0xnm) [inline src/commands/setup/interface.ts:1]: `US` and `EU` are deprecated, customers should use `US1` and `EU1` respectively.

```suggestion
export const datadogSites = ["US1", "EU1", "US3", "US5", "GOV"] as const;
```

File: src/commands/setup/interface.ts [hunk 2]
```diff
@@ -1 +1,5 @@
...
+
+export type DatadogSite = typeof datadogSites[number];
+
+export const DEFAULT_DATADOG_SITE: DatadogSite = "US";
```
Comment (0xnm) [inline src/commands/setup/interface.ts:5]:
```suggestion
export const DEFAULT_DATADOG_SITE: DatadogSite = "US1";
```

File: src/commands/setup/validation.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+const isDatadogSite = (value: unknown): value is DatadogSite => {
+  return datadogSites.includes(value as DatadogSite);
+};
+
+export const validateInitialState = (
+  initialState: SetupCommandInitialStateType
+): SetupCommandStateType => {
+  if (initialState.bypassPrompts && !initialState.apiKey) {
+    throw new ApiKeyNotProvided(initialState.apiKey || "");
```
Comment (0xnm) [inline src/commands/setup/validation.ts:13]: if I get it right, `initialState.apiKey` should be null, or empty, or undefined in this branch. Is it useful to include one of these values into error?

File: src/commands/setup/errors.ts [hunk 4]
```diff
@@ -31,3 +31,23 @@ export class GradlePluginNotAutomated extends Error {
...
+  site: string;
+
+  constructor(site: string, message?: string) {
+    super(message);
+    this.site = site;
+    this.name = "DatadogSiteValueError";
+  }
+}
+
+export class ApiKeyNotProvided extends Error {
```
Comment (0xnm) [inline src/commands/setup/errors.ts:45]:
```suggestion
export class ApiKeyNotProvidedError extends Error {
```

File: src/commands/setup/validation.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  }
+  const datadogSite = initialState.datadogSite;
+  if (datadogSite === undefined) {
+    return {
+      ...initialState,
+      bypassPrompts: !!initialState.bypassPrompts,
+      datadogSite,
+    };
```
Comment (0xnm) [inline src/commands/setup/validation.ts:21]: Shouldn't this be different from return on L27? Shouldn't we put default site if `datadogSite === undefined`?

File: src/commands/setup/create-configuration-files/__tests__/create-configuration-files.test.ts [hunk 6]
```diff
@@ -21,26 +21,42 @@ describe("createConfigurationFiles", () => {
...
         ),
         store
       );
       expect(store.get()).toEqual({
+        bypassPrompts: false,
+        datadogSite: "US",
+      });
+    });
+
+    it("uses the value from the store if", async () => {
```
Comment (0xnm) [inline src/commands/setup/create-configuration-files/__tests__/create-configuration-files.test.ts:37]: Seems like it should be something after `if` here?

**Review by 0xnm — COMMENTED** [review]
I've added some comments. Not sure if all of them make sense though.

File: src/commands/setup/validation.ts (hunk 3 above)
Comment (louiszawadzki) [reply src/commands/setup/validation.ts:13]: Yes I can use `JSON.stringify` here, should give more data

File: src/commands/setup/validation.ts (hunk 5 above)
Comment (louiszawadzki) [reply src/commands/setup/validation.ts:21]: No we should keep it as it is, it will be populated later. However I can refactor this to avoid issues :)

**Review by 0xnm — APPROVED** [review]

**Pull request merged**